docs/source/institutional-faq.rst

```diff
@@ -0,0 +1,362 @@
+your data science users should be able to ask those systems mentioned above for
+resources, and have their processes tracked accordingly.
+
+However, there are institutions where analyst-level users aren't given direct access to
+the cluster.  This is particularly common in Cloudera/Hortonworks Hadoop/Spark deployments.
+In these cases some level of explicit indirection may be required.  For this, we
+recommend the `Dask Gateway <https://gateway.dask.org>`_, which uses IT-level
+permissions to properly route authenticated users into secure resources.
+
+
+How do I manage software environments?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This depends on your cluster resource manager:
+
+-  Most HPC users use their network file system
+-  Hadoop/Spark/Yarn users package their environment into a tarball and ship it
+   around with HDFS (Dask-Yarn integrates with `Conda Pack <TODO>`_ for this
+   capability)
+-  Kubernetes or Cloud users typically use Docker images
+
+In each case Dask typically integrates with existing processes and technologies
+that are well understood and familiar to the institution.
+
+
+For Technical Leads
+-------------------
+
+Will Dask "just work" on our existing code?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+No, you will need to make modifications,
+but these modifications are usually relatively small.
+Certainly the vast majority of lines of business logic within your institution
+will not have to change, assuming that they are in Python and use tooling like
+Numpy, Pandas and Scikit-Learn.
+
+How well does Dask scale?  What are Dask's limitations?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The largest Dask deployments that we see today are on around 1000 multi-core
+machines, perhaps 20,000 cores in total, but these are rare.
+Most institutional-level problems (1-100 TB) are well solved by deployments of 10-50 nodes.
+
+Technically, the back-of-the-envelope number to keep in mind is that each task
+(an individual Python function call) in Dask has an overhead of around *200
+microseconds*.  So if these tasks take 1 second each, then Dask can saturate
+around 5000 cores before scheduling overhead dominates costs.  As workloads
+reach this limit they are encouraged to use larger chunk sizes to compensate.
+The *vast majority* of institutional users though do not reach this limit.
+For more information you may want to peruse our :doc:`best practices
+<best-practices>`
+
+Is Dask Resilient?  What happens when a machine goes down?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, Dask is resilient to the failure of worker nodes.  It knows how it came to
+any result, and can replay the necessary work on other machines if one goes
+down.
+
+If Dask's centralized scheduler goes down then you would need to resubmit the
+computation.  This is a fairly standard level of resiliency today, shared with
+other tooling like Apache Spark, Flink, and others.
```

Comment:
It might be good to mention distributed vs centralized schedulers here?  Ray and [Spark both have a distributed scheduler](https://databricks.com/session/tangram-distributed-scheduling-framework-for-apache-spark-at-facebook), but I don't know much about them.

A concern I've heard is over the centralized scheduler going down, but how likely is this given the dask scheduler isn't doing any of the work itself?  It only has one job.

Reply:
From a brief look, tangram seems to be a distributed system that is happy to launch Spark jobs, rather than an alternative distributed Spark driver.

Ray is genuinely a different architecture.  I'm curious, I know that it uses Redis for metadata communication.  Does it use distributed Redis for this or the more common single-node Redis server, if so, how resilient is distributed Redis?

Reply:
Sorry, I was miss informed about tangram and Spark.  I don't see a native distributed Spark driver.  My internal expert is double checking to see if this is natively possible.  Going back to Spark scheduler resiliency, it is achievable with a resource manager like [Yarn](https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/ResourceManagerHA.html), which also transfers state.

From [this discussion on Ray](https://github.com/ray-project/ray/issues/642#issuecomment-370601221):
> specifications of the tasks are stored in a sharded in-memory database, currently we are not resilient to the failure of this database, but we're prototyping a fault tolerance scheme based on chain replication for it

I believe this is the Redis metadata store you mention, so it is indeed no different from a resilience perspective.  

Is it safe to save the scheduler resilience is left to a resource manager and relying on the user to resubmit computations?

This fact checking was useful since this topic recently came up at C1.  I think what you have is fine, but other readers may want to dig in on this as well.

Reply:
Right, in normal operation relying on a resource manager like Yarn or Kubernetes seems to solve problems for most institutions.  That being said, highly available schedulers have been requested a few times, and is something that could be done.  It's a non-trivial effort though and, despite interest, I haven't seen a case where it was actually necessary.

Reply:
Agreed.  I haven't either, yet people seem to be concerned about it.  I'll keep an eye on it.

Reply:
The stock answer here is that "Dask has the same resilience properties as Spark"